### PR TITLE
Remove Google Analytics driver

### LINKED ISSUE #42792: Remove Google Analytics driver and related special case stuff
We made the decision to deprecate it about a year ago because of https://github.com/metabase/metabase/issues/12905#issuecomment-1611804549, now because it's blocking Metrics v2 we are finally going to drop it entirely. ([We announced this in 47.](https://www.metabase.com/releases/Metabase-47)) This needs to be done before we ship 50. See https://metaboat.slack.com/archives/C04DN5VRQM6/p1715888743074829 for more info.

We need to delete all the code related to GA in the repo including the CI job and any mentions of `:googleanalytics` in the codebase. We have a lot of special edge case stuff in the codebase for GA's hardcoded `:metric`s and `:segment`s (which use string IDs instead of integers), we should remove that stuff from the MBQL schemas and all of the special case code. Examples of stuff to remove:

- `modules/drivers/googleanalytics`
- https://github.com/metabase/metabase/blob/d142da9984b0485c56c224b79116031973226399/src/metabase/legacy_mbql/schema.cljc#L987-L996
- https://github.com/metabase/metabase/blob/59e7b3d1cb2f22a9574cad9aef1a2bea3a72fc80/deps.edn#L388
- https://github.com/metabase/metabase/blob/cd007ba6ac34dc3ab2769c5333718eef41c15d69/src/metabase/xrays/automagic_dashboards/names.clj#L31
- GitHub Actions related to Google Analytics

**Please be thorough and remove all the dead GA-related code!!! Please 🙇‍♂️**

### PR DESCRIPTION
**Pull request by camsaul**
I think I managed to remove all the special case stuff 

Resolves #42792

### CODE REVIEW

File: .github/workflows/drivers-stress-test.yml [hunk 1]
```diff
@@ -109,42 +108,6 @@ jobs:
...
-      DRIVERS: googleanalytics
-    steps:
-    - uses: actions/checkout@v4
-    - name: Test Google Analytics driver
-      uses: ./.github/actions/test-driver
-      with:
-        junit-name: 'be-tests-googleanalytics-ee'
-        test-args: ":only ${{ inputs.test }} :times ${{ inputs.times }}"
-
-  be-google-related-drivers-classpath-test:
```
Comment (camsaul) [inline .github/workflows/drivers-stress-test.yml:127]: this job was to check that Google Analytics and Big Query dependencies didn't conflict with each other so we don't need it anymore

File: .github/workflows/drivers-stress-test.yml [hunk 2]
```diff
@@ -21,7 +21,6 @@ on:
           - Athena
           - BigQuery
           - Druid
-          - Google Analytics
           - Google Classpath Test
```
Comment (lbrdnk) [inline .github/workflows/drivers-stress-test.yml:24]: Google Classpath Test should not be removed?

File: bin/build/src/i18n/enumerate.clj [hunk 3]
```diff
@@ -20,8 +20,7 @@
                             "/enterprise/backend/src"
                             "/modules/drivers/bigquery-cloud-sdk/src"
                             "/modules/drivers/druid/src"
-                            "/modules/drivers/google/src"
-                            "/modules/drivers/googleanalytics/src"
+                            "/modules/drivers/druid-jdbc/src"
```
Comment (lbrdnk) [inline bin/build/src/i18n/enumerate.clj:23]: Thanks!

File: test/metabase/xrays/automagic_dashboards/interesting_test.clj [hunk 4]
```diff
@@ -33,70 +32,6 @@
       (throw (ex-info (format "Did not find %s.%s" (name table) (name column))
                       {:table table :column column}))))
 
-(deftest ^:parallel field-matching-predicates-test
```
Comment (lbrdnk) [inline test/metabase/xrays/automagic_dashboards/interesting_test.clj:36]: Why to remove whole test and not just `ga:` related parts?

**Review by lbrdnk — APPROVED** [review]
LGTM, with comments.

File: test/metabase/xrays/automagic_dashboards/interesting_test.clj (hunk 4 above)
Comment (camsaul) [reply test/metabase/xrays/automagic_dashboards/interesting_test.clj:36]: good catch! Fixed

**Pull request merged**